server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -670,7 +670,7 @@ private void merge(FieldMapper toMerge, Conflicts conflicts) {
             }
         }
 
-        protected void toXContent(XContentBuilder builder, boolean includeDefaults) throws IOException {
+        public void toXContent(XContentBuilder builder, boolean includeDefaults) throws IOException {
```

Comment:
I think you can avoid changing this, the concerns about BWC only apply to existing mappings and won't be a problem for entirely new field mappers.